packages/ra-core/src/dataProvider/performQuery/doQuery.ts

```diff
@@ -28,19 +28,7 @@ export const doQuery = ({
     store,
     mutationMode,
 }: DoQueryParameters) => {
-    const resourceState = store.getState().admin.resources[resource];
-    if (canReplyWithCache(type, payload, resourceState)) {
```

Comment:
I had to disable application cache because it's incompatible with react-query. React-query gives an equivalent, albeit not controlled by the server. It's a BC break we should accept IMO, as application cache is a minor feature not widely used.